src/setup_payload/AdditionalDataPayloadGenerator.h

```diff
@@ -0,0 +1,50 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file provides a utility to generate Additional Data payload and its members
+ *      (e.g. rotating device id)
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPError.h>
+#include <system/TLVPacketBufferBackingStore.h>
+
+#define ROTATING_DEVICE_ID_COUNTER_STR_MAX_LENGTH 10
+#define ROTATING_DEVICE_ID_HASH_SUFFIX_LENGTH 16
+#define ROTATING_DEVICE_ID_MAX_LENGTH ROTATING_DEVICE_ID_COUNTER_STR_MAX_LENGTH + ROTATING_DEVICE_ID_HASH_SUFFIX_LENGTH
+
+namespace chip {
+
+class AdditionalDataPayloadGenerator
+{
+
+public:
+    AdditionalDataPayloadGenerator() {}
+    // Generate additional data payload (i.e. TLV encoded)
+    CHIP_ERROR generateAdditionalDataPayload(uint16_t lifetimeCounter, char * serialNumberBuffer, size_t serialNumberBufferSize,
+                                             chip::System::PacketBufferHandle & bufferHandle, bool enableRotatingDeviceId);
```

Comment:
readability-wise I believe an argument of bitflags for what to enable may be easier to parse. Like:

`generateAdditionalDataPayload(counter, serial, serialSize, packetHandle, true)` is unclear and
`generateAdditionalDataPayload(counter, serial, serialSize, packetHandle, true /* enableRotatingDeviceId *)` is not as extensible as 

`generateAdditionalDataPayload(counter, serial, serialSize, packetHandle, kIncludeRotatingId)`  (in the future, instead of adding more bools, I would add another enum bitfield). This also allows you to ifdef the bitfield value ohn rotating id being enabled.

Reply:
This is obviously nitpicky ... feel free to ignore if it seems too much of a headache.

Reply:
i agree, changed to enum/bitfield